### PR TITLE
parse package-spec from CLI correctly

### LINKED ISSUE #861: Cannot install local packages with bun add (WSL)
### Version

0.1.4

### Platform

Microsoft Windows NT 10.0.22000.0 x64

### What steps will reproduce the bug?

**NOTE**: Since I am on Windows and using WSL, I have included the output from `uname -a` while inside my WSL terminal: `Linux TGDPowerhouse 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`

1) `mkdir a && mkdir b`
2) `cd a && npm init -y && cd ../b && npm init -y`
3) `bun add file://../a`

### How often does it reproduce? Is there a required condition?

Every time

### What is the expected behavior?

`package.json` `dependencies` should be updated to have the local package installed

### What do you see instead?

```shell
b$ bun add file://../a
bun add v0.1.4
  🔍 Resolving [1/1] 
uh-oh: internal: package name mismatch expected "file://../a" but received "a"
bun will crash now 😭😭😭

–––– bun meta ––––
Bun v0.1.4 Linux x64 #1 SMP Wed Mar 2 00:30:59 UTC 2022
AddCommand:
Elapsed: 325ms | User: 385ms | Sys: 256ms
RSS: 73.42MB | Peak: 39.22MB | Commit: 73.42MB | Faults: 0
–––– bun meta ––––

Crash report saved to:
  ~/.bun/.bun-crash/v0.1.4-1658698160262.crash

Ask for #help in https://bun.sh/discord or go to https://bun.sh/issues

thread 294 panic: internal: package name mismatch expected "file://../a" but received "a"
???:?:?: 0x47f6959 in ??? (???)
???:?:?: 0x4784e38 in ??? (???)
???:?:?: 0x4783da1 in ??? (???)
???:?:?: 0x4783a31 in ??? (???)
???:?:?: 0x43b2949 in ??? (???)
???:?:?: 0x47dd572 in ??? (???)
???:?:?: 0x7f2f0294d608 in ??? (???)
Aborted
b$ 
```

### Additional information

[Asked to file by Jarred in the Discord channel](https://discord.com/channels/876711213126520882/887787428973281300/1000874701767311460)

Can provide add'l information as requested

### PR DESCRIPTION
**Pull request by alexlamsl**
fixes #861

### CODE REVIEW

File: src/install/install.zig [hunk 1]
```diff
@@ -5086,85 +5086,77 @@ pub const PackageManager = struct {
...
+                        };
+                    },
+                    else => {},
                 }
 
-                request.name = std.mem.trim(u8, request.name, "\n\r\t");
-                if (request.name.len == 0) continue;
-
+                var request = UpdateRequest{
+                    .name = switch (version.tag) {
```
Comment (Jarred-Sumner) [inline src/install/install.zig]: this is a pointer to a temporary which becomes invalid after this stack frame (scope)

Comment (alexlamsl) [reply src/install/install.zig]: `String` is temporary, yes, but it is immediately `.slice(value)` into `[] u8`, and `value` is allocated on the heap AFAICT...? :thinking:

File: src/install/install.zig [hunk 2]
```diff
@@ -5086,85 +5086,77 @@ pub const PackageManager = struct {
...
+                        .dist_tag => version.value.dist_tag.name,
+                        .github => version.value.github.repo,
+                        .npm => version.value.npm.name,
+                        .symlink => version.value.symlink,
+                        else => version.literal,
+                    }.slice(value),
```
Comment (alexlamsl) [inline src/install/install.zig]: ... as in right here, it is being converted from `String` (stack) to `string` (heap?)

Comment (Jarred-Sumner) [reply src/install/install.zig]: if the name/version is <= 8 characters, the pointer will be to the inline `String` instead of the heap

Comment (alexlamsl) [reply src/install/install.zig]: Done in https://github.com/oven-sh/bun/pull/1895/commits/cd3aa41a203b0876ee9ab9423856d8833af18140

File: src/install/install.zig (hunk 1 above)
Comment (alexlamsl) [reply src/install/install.zig]: Done in https://github.com/oven-sh/bun/pull/1895/commits/cd3aa41a203b0876ee9ab9423856d8833af18140

**Comment by Jarred-Sumner** [comment]
thank you

**Comment by EvanBacon** [comment]
This should also apply to `resolutions` and `overrides`, e.g. 
```json
"resolutions": {
    "foo": "./foo.tgz",
}
```

**Pull request merged**